### PR TITLE
BaSP-TG-99: Admins redux - Create, update and delete

### PR DESCRIPTION
**Pull request by MaicolFreire001**
**Admis:**

Create:
- Created constants to create an admin
- Created actions to create admin
- Created thunk to create admin
- Completed admins reducer with the logic to create
- Connected form component with store to create an admin

Update:
- Created constants to update an admin
- Created actions to update admin
- Created thunk to update admin
- Completed admins reducer with the logic to update an admin
- Connected form component with store to update an admin

Delete :
- Created constants to delete an admin
- Created actions to delete admin
- Created thunk to delete admin
- Completed admins reducer with the logic to delete an admin
- Connected form component with store to delete an admin

### CODE REVIEW

File: src/redux/admins/reducer.js [hunk 1]
```diff
@@ -21,12 +34,69 @@ const reducer = (state = INITIAL_STATE, action) => {
...
+        list: [...state.list, action.data],
+        error: '',
+        isLoading: false
+      };
+    case POST_ADMINS_ERROR:
       return {
         ...state,
         isLoading: false,
         error: action.error,
         list: []
```
Comment (ignacioaristo) [inline src/redux/admins/reducer.js:60]: The list into the initial state is already an empty array

File: src/redux/admins/reducer.js [hunk 2]
```diff
@@ -21,12 +34,69 @@ const reducer = (state = INITIAL_STATE, action) => {
...
+        list: [...state.list, action.data],
+        error: '',
+        isLoading: false
+      };
+    case PUT_ADMINS_ERROR:
+      return {
+        ...state,
+        isLoading: false,
+        error: action.payload,
+        list: []
```
Comment (ignacioaristo) [inline src/redux/admins/reducer.js:79]: Remove all.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/redux/admins/reducer.js (hunk 1 above)
Comment (MaicolFreire001) [reply src/redux/admins/reducer.js:60]: Resolved

File: src/redux/admins/reducer.js (hunk 2 above)
Comment (MaicolFreire001) [reply src/redux/admins/reducer.js:79]: Done

**Review by GonzaloAsencio — APPROVED** [review]

File: src/redux/admins/thunks.js [hunk 3]
```diff
@@ -17,3 +30,90 @@ export const getAdmins = () => {
...
+      body: JSON.stringify({
+        name: data.name,
+        lastName: data.lastName,
+        email: data.email,
+        password: data.password
+      })
+    })
+      .then((response) => response.json())
+      .then((response) => {
+        if (response.error) {
```
Comment (fede230298) [inline src/redux/admins/thunks.js:80]: Change the second .then response to data, in order to not repeat names and be clearer.

Repeat in all cases

**Review by fede230298 — APPROVED** [review]

File: src/redux/admins/thunks.js (hunk 3 above)
Comment (MaicolFreire001) [reply src/redux/admins/thunks.js]: Done

**Review by ignacioaristo — APPROVED** [review]

File: src/redux/admins/reducer.js [hunk 4]
```diff
@@ -24,8 +37,61 @@ const reducer = (state = INITIAL_STATE, action) => {
...
+      };
+    case PUT_ADMINS_PENDING:
+      return {
+        ...state,
+        isLoading: true
+      };
+    case PUT_ADMINS_SUCCESS:
+      return {
+        ...state,
+        list: [...state.list, action.data],
```
Comment (fede230298) [inline src/redux/admins/reducer.js:68]: You are adding a new item to the list. Lets change the item with .map

Comment (Ignacio-Sormani) [inline src/redux/admins/reducer.js:68]: Use map method to update the list

File: src/redux/admins/actions.js [hunk 5]
```diff
@@ -1,4 +1,17 @@
-import { GET_ADMINS_PENDING, GET_ADMINS_SUCCESS, GET_ADMINS_ERROR } from './constant';
+import {
+  GET_ADMINS_PENDING,
```
Comment (Ignacio-Sormani) [inline src/redux/admins/actions.js:2]: When you make a post, put or delete you are making changes into a single admin, so actions, constants, thunks, etc should be in singular. For example: POST_ADMIN_PENDING

Comment (MaicolFreire001) [reply src/redux/admins/actions.js:2]: Done

File: src/redux/admins/reducer.js (hunk 4 above)
Comment (MaicolFreire001) [reply src/redux/admins/reducer.js:68]: Done

File: src/redux/admins/actions.js (hunk 5 above)
Comment (MaicolFreire001) [reply src/redux/admins/actions.js:2]: Done

File: src/redux/admins/reducer.js [hunk 6]
```diff
@@ -24,8 +37,61 @@ const reducer = (state = INITIAL_STATE, action) => {
...
+      };
+    case DELETE_ADMIN_PENDING:
+      return {
+        ...state,
+        isLoading: true
+      };
+    case DELETE_ADMIN_SUCCESS:
+      return {
+        ...state,
+        list: [...state.list.filter((task) => task._id !== action.payload)],
```
Comment (ProDIGGY-vinilos) [inline src/redux/admins/reducer.js:86]: Let's update to use .map

File: src/redux/admins/reducer.js (hunk 4 above)
Comment (Ignacio-Sormani) [reply src/redux/admins/reducer.js:68]: I don't see this is resolved. Take a look at the PR that fede made and that is already merged

**Review by Ignacio-Sormani — CHANGES_REQUESTED** [review]

File: src/redux/admins/reducer.js (hunk 4 above)
Comment (MaicolFreire001) [reply src/redux/admins/reducer.js]: You're right, I forgot to save. It's pushed now

**Review by Ignacio-Sormani — APPROVED** [review]

**Review by sofihoj — APPROVED** [review]

**Pull request merged**